test/test_metrics.rb

```diff
@@ -104,16 +99,19 @@ def create_known_metrics(time = fixed_time)
     it "fetches top job data" do
       create_known_metrics
       q = Sidekiq::Metrics::Query.new(now: fixed_time)
-      rs = q.top_jobs
-      assert_equal Date.new(2022, 7, 22), rs[:date]
-      assert_equal 2, rs[:job_classes].size
-      assert_equal "App::SomeJob", rs[:job_classes].first
-      bucket = rs[:totals]
-      refute_nil bucket
-      assert_equal bucket.keys.sort, ["App::FooJob|ms", "App::FooJob|p", "App::SomeJob|f", "App::SomeJob|ms", "App::SomeJob|p"]
-      assert_equal 3, bucket["App::SomeJob|p"]
-      assert_equal 4, bucket["App::FooJob|p"]
-      assert_equal 1, bucket["App::SomeJob|f"]
+      result = q.top_jobs
+      assert_equal fixed_time - 59 * 60, result.starts_at
+      assert_equal fixed_time, result.ends_at
+
+      assert_equal 60, result.buckets.size
+      assert_equal "21:04", result.buckets.first
+      assert_equal "22:03", result.buckets.last
+
+      assert_equal %w[App::SomeJob App::FooJob].sort, result.job_results.keys.sort
+      some_job_result = result.job_results["App::SomeJob"]
+      refute_nil some_job_result
+      assert_equal 1, some_job_result.series.dig("ms", "22:03")
```

Comment:
This test is randomly failing with different actual values, and I'm not sure why, since we're using a fixed time. It passes most of the time.

```
Failure:
Sidekiq::Metrics::querying#test_0002_fetches top job data [test/test_metrics.rb:118]:
Expected: 1
  Actual: 8

Failure:
Sidekiq::Metrics::querying#test_0002_fetches top job data [test/test_metrics.rb:118]:
Expected: 1
  Actual: 2
```

Reply:
Could it help to debug it by adding the whole hash as the message?

```suggestion
      assert_equal 1, some_job_result.series.dig("ms", "22:03"), some_job_result
```

Reply:
Fixed in https://github.com/mperham/sidekiq/pull/5467/commits/d0b1616ad82250204cf1a270aac17a64000de37b.